### PR TITLE
Update and Integrate ABI

### PR DESCRIPTION
**Pull request by VoR0220**
fixes the following:

https://github.com/eris-ltd/eris-cli/issues/1040
https://github.com/eris-ltd/eris-cli/issues/1026
https://github.com/eris-ltd/eris-cli/issues/1002
https://github.com/eris-ltd/eris-cli/issues/1001
https://github.com/eris-ltd/eris-cli/issues/1000

### CODE REVIEW

File: tests/test.sh [hunk 1]
```diff
@@ -142,7 +142,7 @@ run_test(){
     echo
     cat readme.md
     echo
-    eris pkgs do --chain "$chain_name" --address "$key1_addr" --set "addr1=$key1_addr" --set "addr2=$key2_addr" --set "addr2_pub=$key2_pub" #--debug
+    eris pkgs do --chain "$chain_name" --address "$key1_addr" --set "addr1=$key1_addr" --set "addr2=$key2_addr" --set "addr2_pub=$key2_pub" --debug
```
Comment (zramsay) [inline tests/test.sh]: hmm

Comment (VoR0220) [inline tests/test.sh]: temporary...going to get rid of it when it's working.

**Comment by VoR0220** [comment]
https://github.com/ethereum/go-ethereum/pull/3121

**Comment by benjaminbollen** [comment]
@VoR0220 the tests are not run in circle; see build of #251

**Comment by benjaminbollen** [comment]
LGTM

**Pull request merged**